p4-16/psa/PSA.mdk

```diff
@@ -582,6 +673,138 @@ extern DirectCounter<W> {
 }
 ```
 
+A `DirectCounter` instance must appear in the list of values of the
+`psa_direct_counters` table attribute for exactly one table.  We call
+this table the `DirectCounter` instance's "owner".  It is illegal to
+call the `count` method for a `DirectCounter` instance anywhere except
+inside an action of its owner table.
+
+The counter value updated by an invocation of `count` is always the
+one associated with the table entry that matched.
+
+An action of an owner table need not have `count` method calls for all
+of the `DirectCounter` instances that the table owns.
+
+The reference implementation for the `DirectCounter` extern is
+essentially the same as that for `Counter`, including the saturating
+behavior.  Since there is no `index` parameter to the `count` method,
+there is no need to check for whether it is in range.
+
+TBD: What to say about `entry_key` type?  Does it need to be made more
+precise?
```

Comment:
I believe this is the issue that @jnfoster brought up yesterday: the entry_key must match the table to which the counter is associated, and therefore we can't have a single entry key defined for the whole PSA. He has proposed wording yesterday, something to this effect: PSA should specify that the counters/meters will use the P4 runtime mode of addressing